com.unity.sg2/Editor/GraphUI/DataModel/GraphDataBlockNodeModel.cs

```diff
@@ -0,0 +1,74 @@
+using System.Linq;
+using Unity.GraphToolsFoundation;
+using Unity.GraphToolsFoundation.Editor;
+using UnityEditor.ShaderGraph.GraphDelta;
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph.GraphUI
+{
+    class GraphDataBlockNodeModel : BlockNodeModel, IGraphDataOwner
+    {
+        [field: SerializeField, HideInInspector] public string ContextEntryName { get; set; }
+
+        protected override PortModel CreatePort(PortDirection direction,
+            PortOrientation orientation,
+            string portName,
+            PortType portType,
+            TypeHandle dataType,
+            string portId,
+            PortModelOptions options)
+        {
+            return new GraphDataPortModel(this, direction, orientation, portName ?? "", portType, dataType, portId, options);
+        }
+
+        protected override void OnDefineNode()
+        {
+            if (ContextNodeModel is not GraphDataContextNodeModel graphDataContext)
+            {
+                return;
+            }
+
+            if (!graphDataContext.TryGetNodeHandler(out var nodeHandler))
+            {
+                return;
+            }
+
+            var portHandler = nodeHandler.GetPort(ContextEntryName);
```

Comment:
Similar to my above comment, would it be possible to remove these references to CLDS concepts by using the UI-side view model information we have now?